search.go

```diff
@@ -0,0 +1,83 @@
+package onedrive
+
+import (
+	"context"
+)
+
+type EntityType string
+
+// DriveItem declares to search drive items
+const DriveItem EntityType = "driveItem"
+
+// SearchService allows searching almost any resource
+type SearchService interface {
+	Search(ctx context.Context, query SearchRequest) (*SearchResponse, error)
+}
+
+type searchService struct {
+	client *OneDrive
+}
+
+type queryPayload struct {
+	Query string `json:"queryString"`
+}
+
+type SearchRequests struct {
+	Requests []SearchRequest `json:"requests"`
+}
+
+type SearchRequest struct {
+	EntityTypes []EntityType `json:"entityTypes"`
+	Query       queryPayload `json:"query"`
```

Comment:
rather than using the private type here, you could accept `SearchOptions` and give it a custom `MarshalJSON()` - or even just have this client call `QueryPayload`

Reply:
Ah yes I like the MarshalJSON approach 👍 will update